### PR TITLE
[chore] remove deprecated use of confmap.WithErrorUnused

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: receiver/gitproviderreceiver/config.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
 	}
 
 	// load the non-dynamic config normally
-	err := componentParser.Unmarshal(cfg)
+	err := componentParser.Unmarshal(cfg, confmap.WithIgnoreUnused())
```
Comment (atoulme) [inline receiver/gitproviderreceiver/config.go:47]: Migrating here to make it explicit - the default behavior is now to error out on unused elements.

File: connector/countconnector/config.go [hunk 2]
```diff
@@ -130,7 +130,7 @@ func (c *Config) Unmarshal(componentParser *confmap.Conf) error {
 		// Nothing to do if there is no config given.
 		return nil
 	}
-	if err := componentParser.Unmarshal(c); err != nil {
+	if err := componentParser.Unmarshal(c, confmap.WithIgnoreUnused()); err != nil {
```
Comment (atoulme) [inline connector/countconnector/config.go:133]: Migrating to explicitly call out this unmarshaling ignores unused elements. The default behavior has changed to error out on unused elements.

File: exporter/signalfxexporter/config.go [hunk 3]
```diff
@@ -202,7 +202,7 @@ func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
 		return nil
 	}
 
-	if err := componentParser.Unmarshal(cfg); err != nil {
+	if err := componentParser.Unmarshal(cfg, confmap.WithIgnoreUnused()); err != nil {
```
Comment (atoulme) [inline exporter/signalfxexporter/config.go:205]: Migrating to explicitly call out this unmarshaling ignores unused elements. The default behavior has changed to error out on unused elements.

**Review by songy23 — APPROVED** [review]
LGTM

**Review by djaglowski — APPROVED** [review]

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**